### PR TITLE
[CPLD-1730] Refactor deferring a participant

### PR DESCRIPTION
**Pull request by cwrw**
### Context

To move to the new world of API rendering and services, refactor Defer API endpoint.

- Ticket: https://dfedigital.atlassian.net/jira/software/projects/CPDLP/boards/87?selectedIssue=CPDLP-1730

### Changes proposed in this pull request
- Add new defer participant service
- Add methods between NPQ and ECF profiles to expose a shared interface, to simplify the service
- Use new API methods to surface response or validation errors
- Remove old defer participant implementation

### Guidance to review

Since we have to live in both worlds of `participant_actions` vs new actions, we will need to reuse some of those components such as `params_for_recorder` etc. however when we move all actions to the new world, we can get rid of participant actions and have have explicit methods for both NPQ and ECF

### CODE REVIEW

File: app/controllers/api/v1/npq_participants_controller.rb
Comment (cwrw) [inline app/controllers/api/v1/npq_participants_controller.rb:33]: Since we have to live in both worlds of `participant_actions` vs new actions, we will need to reuse some of those components such as `params_for_recorder` etc. however when we move all actions to the new world, we can get rid of participant actions and have have explicit methods for both NPQ and ECF

**Review by leandroalemao — APPROVED** [review]

File: app/models/participant_profile/ecf.rb
Comment (peteryates) [reply app/models/participant_profile/ecf.rb:107]: I'd agree the method name doesn't look like it's lining up with the intent. It's also just wrapping `#relevant_induction_record` - could it be an alias?

**Review by peteryates — APPROVED** [review]
This is a **huge** improvement. The deferral process is already way easier to follow.

**Comment by cwrw** [comment]
@peteryates @StupidCodeFactory @leandroalemao thanks for your review. I've addressed all comments in a separate un-blocker PR: https://github.com/DFE-Digital/early-careers-framework/pull/2627 as there are multiple refactors happening at once, to unblock others while this is tested, if you don't mind having a look there. I'll rebase this as soon as that is merged. Thanks!

File: app/services/defer_participant.rb
Comment (leandroalemao) [inline app/services/defer_participant.rb:48]: Is this being used anywhere?

**Review by StupidCodeFactory — APPROVED** [review]

File: app/services/defer_participant.rb
Comment (cwrw) [reply app/services/defer_participant.rb:48]: We need the surface schedule for the induction record validator, see: https://github.com/DFE-Digital/early-careers-framework/blob/develop/app/validators/induction_record_validator.rb#L31

**Pull request merged**
